beta-src/src/components/ui/WDFullModal.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from "react";
+import { useState } from "react";
+import { Box, Stack, Button } from "@mui/material";
+import WDInfoDisplay from "./WDInfoDisplay";
+import GameMode from "../../enums/GameMode";
+import Season from "../../enums/Season";
+import Ranking from "../../enums/Ranking";
+import IntegerRange from "../../types/IntegerRange";
+import GameType from "../../enums/GameType";
+import { CountryTableData } from "../../interfaces";
+import Device from "../../enums/Device";
+import WDInfoPanel from "./WDInfoPanel";
+import WDPress from "./WDPress";
+import ModalViews from "../../enums/ModalViews";
+
+interface WDFullModalProps {
+  children: React.ReactNode;
+  countries: CountryTableData[];
+  device: Device;
+  gameMode: GameMode;
+  gameType: GameType;
+  maxDelays: IntegerRange<0, 5>;
+  potNumber: IntegerRange<35, 665>;
+  rank: Ranking;
+  season: Season;
+  title: string;
+  userCountry: CountryTableData;
+  year: number;
+}
+
+const textButtonStyle = {
+  borderRadius: 0,
+  fontWeight: 400,
+  minWidth: 0,
+  p: "0 0 5px 0",
+  "&:hover": {
+    background: "transparent",
+  },
+};
+
+const textButtonSelected = {
+  borderRadius: 0,
+  borderBottom: "solid black 2px",
+  fontWeight: 600,
+  minWidth: 0,
+  p: "0 0 5px 0",
+  "&:hover": {
+    background: "transparent",
+  },
+};
+
+const WDFullModal: React.FC<WDFullModalProps> = function ({
+  children,
+  countries,
+  device,
+  gameMode,
+  gameType,
+  maxDelays,
+  potNumber,
+  rank,
+  season,
+  title,
+  userCountry,
+  year,
+}): React.ReactElement {
+  const [view, setView] = useState("info");
+  const mobileLandscapeLayout =
```

Comment:
I will sync up this suggestion with Bryan since he created the getDevice help function. For what I can see now, getDevice is mainly for WDMapController and the scales are determined by different device... 
```

const Scales: Scale = {
  DESKTOP: [0.45, 3],
  MOBILE_LG: [0.32, 1.6],
  MOBILE_LG_LANDSCAPE: [0.3, 1.6],
  MOBILE: [0.32, 1.6],
  MOBILE_LANDSCAPE: [0.27, 1.6],
  TABLET: [0.6275, 3],
  TABLET_LANDSCAPE: [0.6, 3],
};
```
if we add one enum option for isMobilLandscape, that means the new option will affect both MOBILE_LANDSCAPE and MOBILE_LG_LANDSCAPE

```
const mobileLandscapeLayout =
    device === Device.MOBILE_LANDSCAPE || device === Device.MOBILE_LG_LANDSCAPE;
```

Reply:
ok i am just thinking of a why to dry up this code because we are repeating it a lot...i see why now, but maybe it needs to be a hook that gives us the device information and returns isMobilLandscape for us